internal/sbi/processor/influence_data_subscriptions_collection.go

```diff
@@ -0,0 +1,140 @@
+/*
+ * Nudr_DataRepository API OpenAPI file
+ *
+ * Unified Data Repository Service
+ *
+ * API version: 1.0.0
+ * Generated by: OpenAPI Generator (https://openapi-generator.tech)
+ */
+
+package processor
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+
+	"github.com/gin-gonic/gin"
+	"go.mongodb.org/mongo-driver/bson"
+
+	"github.com/free5gc/openapi/models"
+	udr_context "github.com/free5gc/udr/internal/context"
+	"github.com/free5gc/udr/internal/logger"
+	"github.com/free5gc/udr/internal/util"
+	"github.com/free5gc/util/mongoapi"
+)
+
+func (p *Processor) ApplicationDataInfluenceDataSubsToNotifyGetProcedure(
+	c *gin.Context, dnn string, snssai *models.Snssai, internalGroupId, supi string,
+) {
+	var response []models.TrafficInfluSub
+
+	udrSelf := udr_context.GetSelf()
+	udrSelf.InfluenceDataSubscriptions.Range(func(key, value interface{}) bool {
+		subs, ok := value.(*models.TrafficInfluSub)
+		if !ok {
+			logger.DataRepoLog.Errorf("Failed to load influence Data subscription ID [%+v]", key)
+			return true
+		} else if dnn != "" && !util.Contain(dnn, subs.Dnns) {
+			return true
+		} else if snssai != nil && !util.Contain(*snssai, subs.Snssais) {
+			return true
+		} else if internalGroupId != "" && !util.Contain(internalGroupId, subs.InternalGroupIds) {
+			return true
+		} else if supi != "" && !util.Contain(supi, subs.Supis) {
+			return true
+		} else {
+			response = append(response, *subs)
+		}
+		return true
+	})
```

Comment:
Why this function return true

Reply:
This is a function passed to the "Range" method, so the return value is not actually the return value of "ApplicationDataInfluenceDataSubsToNotifyGetProcedure".

Reply:
Always returning "true" here means continuing to iterate over the "sync.Map". When the "else" condition is met, the value is added to the "response".